azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```diff
@@ -71,24 +77,88 @@
 
     @Override
     protected void doExecute() throws AzureExecutionException {
-        // todo: use parser to getAzureClient from mojo configs
-        try {
-            final RuntimeHandler runtimeHandler = getFactory().getRuntimeHandler(
-                    getWebAppConfiguration(), getAzureClient());
-            // todo: use parser to get web app from mojo configs
-            final WebApp app = getWebApp();
-            if (app == null) {
-                if (this.isDeployToDeploymentSlot()) {
-                    throw new AzureExecutionException(WEBAPP_NOT_EXIST_FOR_SLOT);
-                }
-                createWebApp(runtimeHandler);
-            } else {
-                updateWebApp(runtimeHandler, app);
-            }
-            deployArtifacts(getWebAppConfiguration());
-        } catch (IOException | AzureAuthFailureException | InterruptedException e) {
-            throw new AzureExecutionException(
-                    String.format("Encoutering error when deploying to azure: '%s'", e.getMessage()), e);
+        final WebAppConfig config = getWebAppConfig();
+        final IAppService target = createOrUpdateResource(config);
+        deployArtifacts(target, config);
+    }
+
+    private IAppService createOrUpdateResource(final WebAppConfig config) throws AzureExecutionException {
+        final IAppService target = getDeployTarget(config);
+        if (target instanceof IWebApp) {
+            return target.exists() ? updateWebApp((IWebApp) target, config) : createWebApp((IWebApp) target, config);
+        } else {
+            return target.exists() ? updateDeploymentSLot((IWebAppDeploymentSlot) target, config) :
+                    createDeploymentSlot((IWebAppDeploymentSlot) target, config);
+        }
+    }
+
+    private IAppService getDeployTarget(final WebAppConfig config) throws AzureExecutionException {
+        final IWebApp webApp = getAppServiceClient().webapp(config.getResourceGroup(), config.getAppName());
+        final boolean isDeploymentSlot = StringUtils.isNotEmpty(config.getDeploymentSlotName());
+        if (isDeploymentSlot && !webApp.exists()) {
+            throw new AzureExecutionException(WEBAPP_NOT_EXIST_FOR_SLOT);
+        }
+        return isDeploymentSlot ? webApp.deploymentSlot(config.getDeploymentSlotName()) : webApp;
+    }
+
+    private IWebApp createWebApp(final IWebApp webApp, final WebAppConfig webAppConfig) {
+        // todo: get or create Resource Group
+        final ResourceGroup resourceGroup = null;
+        // Get or create App Service Plan
+        final IAppServicePlan appServicePlan = getAppServiceClient()
```

Comment:
the method name of `AzureAppService getAppServiceClient()` get me confused.
can you make `AzureAppService` a static field named `az` or `appService` or some other.